### PR TITLE
Cleanup GrpcConnectionPool constructor

### PR DESCRIPTION
**Pull request by rongrong**

### CODE REVIEW

File: core/common/src/main/java/alluxio/grpc/GrpcConnectionPool.java [hunk 1]
```diff
@@ -38,6 +41,9 @@
 import java.util.concurrent.atomic.AtomicLong;
 import javax.annotation.concurrent.ThreadSafe;
 
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
```
Comment (LuQQiu) [inline core/common/src/main/java/alluxio/grpc/GrpcConnectionPool.java:44]: import static in front of normal import?

File: core/common/src/main/java/alluxio/grpc/GrpcConnectionPool.java [hunk 2]
```diff
@@ -51,40 +57,21 @@ public class GrpcConnectionPool {
...
-   * @param conf Alluxio configuration
-   * @return Ssl context provider instance
-   */
-  private synchronized SslContextProvider getSslContextProvider(AlluxioConfiguration conf) {
-    if (mSslContextProvider == null) {
-      mSslContextProvider = SslContextProvider.Factory.create(conf);
-    }
-    return mSslContextProvider;
-  }
+  private final SslContextProvider mSslContextProvider =
```
Comment (LuQQiu) [inline core/common/src/main/java/alluxio/grpc/GrpcConnectionPool.java:73]: Do we always want to initialize the SslContextProvider?
Looks like SslContextProvider is not always needed.
Will it be expensive in some cases (for some SSL context other than the default one which does nothing)

**Review by LuQQiu — COMMENTED** [review]
Thanks for refactoring, left some comments

File: core/common/src/main/java/alluxio/grpc/GrpcConnectionPool.java (hunk 2 above)
Comment (LuQQiu) [reply core/common/src/main/java/alluxio/grpc/GrpcConnectionPool.java:73]: Double checked, it's not expensive, just storing the conf

**Review by LuQQiu — APPROVED** [review]
LGTM except minor import comment

**Comment by rongrong** [comment]
alluxio-bot, merge this please

**Pull request merged**